packages/frontend-2/lib/common/helpers/graphql.ts

```diff
@@ -499,52 +529,182 @@ export const getDateCursorFromReference = (params: {
 }
 
 /**
- * Simplified version of modifyObjectFields, just targetting a single field
+ * Build skipLoggingErrors function that skips logging errors if there's only one error and it's related to a specific field
+ */
+export const skipLoggingErrorsIfOneFieldError =
+  (fieldName: string | string[]) =>
+  (err: ErrorResponse): boolean => {
+    const fieldNames = isArray(fieldName) ? fieldName : [fieldName]
+    return (
+      err.graphQLErrors?.length === 1 &&
+      err.graphQLErrors.some((e) => intersection(e.path || [], fieldNames).length > 0)
+    )
+  }
+
+type NonUndefined<T> = T extends undefined ? never : T
+
+/**
+ * Update field at specific path in object, only if it exists. Useful for cache modification
+ * when fields should only be updated if they exist.
+ */
+export const updatePathIfExists = <Value, Path extends Paths<Value> & string>(
+  val: Value,
+  path: Path,
+  updater: (val: NonUndefined<Get<Value, Path>>) => NonUndefined<Get<Value, Path>>
+) => {
+  if (!val) return val
+
+  if (has(val, path)) {
+    const pathVal = get(val, path) as NonUndefined<Get<Value, Path>>
+    const newVal = updater(pathVal)
+    set(val, path, newVal)
+  }
+
+  return val
+}
+
+/**
+ * Get value from specific path in object, only if it exists.
+ */
+export const getFromPathIfExists = <Value, Path extends Paths<Value> & string>(
+  val: MaybeNullOrUndefined<Value>,
+  path: Path
+): Optional<Get<Value, Path>> => {
+  if (!val) return undefined
+  if (!has(val, path)) return undefined
+  return get(val, path) as Get<Value, Path>
+}
+
+type ModifyObjectFieldValue<
+  Type extends keyof AllObjectTypes,
+  Field extends keyof AllObjectTypes[Type]
+> = ModifyFnCacheData<AllObjectTypes[Type][Field]>
+
+/**
+ * Simplified & improved version of modifyObjectFields, just targetting a single field for a cache modification
  * @see modifyObjectFields
  */
 export const modifyObjectField = <
-  FieldData = unknown,
-  Variables extends Optional<Record<string, unknown>> = undefined
+  Type extends keyof AllObjectTypes,
+  Field extends keyof AllObjectTypes[Type]
 >(
   cache: ApolloCache<unknown>,
-  id: string,
-  fieldName: string,
+  key: ApolloCacheObjectKey<Type>,
+  fieldName: Field,
   updater: (params: {
     fieldName: string
-    variables: Variables
-    value: ModifyFnCacheData<FieldData>
-    details: Parameters<Modifier<ModifyFnCacheData<FieldData>>>[1] & {
+    variables: Field extends keyof AllObjectFieldArgTypes[Type]
+      ? AllObjectFieldArgTypes[Type][Field]
+      : never
+    /**
+     * Value found in the cache. Read-only and should not be mutated directly. Use the
+     * createUpdatedValue() helper to build a new value with updated fields.
+     */
+    value: ReadonlyDeep<ModifyObjectFieldValue<Type, Field>>
+    helpers: {
```

Comment:
Access all of the helpers you need from the `helpers` object in the modifyObjectField() callback